src/Store/SyntaxHighlightingStoreConnector.re

```diff
@@ -53,18 +53,18 @@ let start = (cli: Core.Cli.t, languageInfo: Ext.LanguageInfo.t) => {
       |> Core.Configuration.getValue(c => c.syntaxEagerMaxLineLength);
 
     let len = min(Array.length(lines), maxLines);
-    let idx = ref(0);
-    let limitExceeded = ref(false);
 
-    while (idx^ < len && ! limitExceeded^) {
-      if (String.length(lines[idx^]) > maxLineLength) {
-        limitExceeded := true;
-      };
-
-      incr(idx);
+    let rec loop = (idx) => {
+      if (idx >= len) {
+        idx
+      } else if (String.length(lines[idx]) > maxLineLength) {
+        idx;
+      } else {
+        loop(idx + 1)
+      }
     };
 
-    let numberOfLinesToHighlight = idx^;
+    let numberOfLinesToHighlight = loop(0);
```

Comment:
Nice. This looks so much clearer to me.

You can use a scope here to prevent the `loop` function from polluting the namespace unnecessarily:

```reason
let numberOfLinesToHighlight = {
  let rec loop = ...

  loop(0);
};